vnet/lib/vnet/node_api/datapath_generic.rb

```diff
@@ -37,6 +37,41 @@ def find_ip_lease_id(interface_id)
 
   class DatapathNetwork < DatapathGeneric
     class << self
+      def associate_network(datapath_uuid, network_uuid, interface_uuid, broadcast_mac_address)
+        transaction do
+          datapath = Vnet::Models::Datapath[datapath_uuid]
+          network = Vnet::Models::Network[network_uuid]
+          interface = Vnet::Models::Interface[interface_uuid]
+          if broadcast_mac_address.nil?
+            broadcast_mac_address = generate_new_mac_address
+          end
+
+          Vnet::Models::DatapathNetwork.create({datapath_id: datapath.id,
+                                                 interface_id: interface.id,
+                                                 network_id: network.id,
+                                                 broadcast_mac_address: broadcast_mac_address
+                                               })
```

Comment:
The direct create call on model completely circumvents the event handling done by EventBase, you should use the create of the class.

Using UUID here seems rather weird, we already query the id's needed so a straightforward create call with a missing broadcast_mac_address should be enough.

https://github.com/axsh/openvnet/blob/feature-dpnetwork-auto-assign/vnet/lib/vnet/node_api/interface.rb#L65

The create_with_transaction method can be overloaded, and the transaction placed in that method.

https://github.com/axsh/openvnet/blob/feature-dpnetwork-auto-assign/vnet/lib/vnet/node_api/datapath_generic.rb#L9

DatapathGeneric has a create_with_transaction method already, however it is missing the transaction block, and should be fixed. You should add the broadcast mac address support using this path.